src/compass/utils/ale.py

```diff
@@ -0,0 +1,325 @@
+
+    if plot_ale:
+        fig, ax = plt.subplots(figsize=(8, 6))
+        sc = ax.scatter(cr_df['ALE_Y'], cr_df['ALE_X'], s=200, alpha=0.8,
+                        marker='o')
+
+        ax.grid(True)
+        ax.set_xlim(-10, 10)
+        ax.set_ylim(-10, 10)
+        ax.axhline(0, color='black')
+        ax.axvline(0, color='black')
+        ax.set_xlabel('Easting error (m)')
+        ax.set_ylabel('Northing error (m)')
+        fig.suptitle('Absolute geolocation error')
+        plt.show()
+
+
+def find_peak(arr, x_loc, y_loc, ovs_factor=64, margin=8):
+    '''
+    Find peak location in 'arr'
+    Parameters
+    ----------
+    arr: np.ndarray
+        Array to use for identifying peak location
+    x_loc: np.int
+        First guess of peak location along X-coordinates
+    y_loc: np.int
+        First guess of peak location along Y-coordinates
+    ovs_factor: np.int
+        Oversampling factor
+    margin: int
+        Margin
+
+    Returns
+    -------
+    x_peak: np.float
+        Peak location along X-coordinates
+    y_peak: np.float
+        Peak location along Y-coordinate
+    '''
+
+    # Extract an area around x_loc, y_loc
+    img = arr[(y_loc - margin):(y_loc + margin),
+              (x_loc - margin):(x_loc + margin)]
+
+    # Oversample CSLC subset and get amplitude
+    img_ovs = isce3.cal.point_target_info.oversample(
+        img, ovs_factor)
+    idx_peak_ovs = np.argmax(np.abs(img_ovs))
+    img_peak_ovs = np.unravel_index(idx_peak_ovs, img_ovs.shape)
+
+    # Get location of the peak wrt the upper left corner of image subset
+    x_peak = x_loc - margin + img_peak_ovs[1] / ovs_factor
+    y_peak = y_loc - margin + img_peak_ovs[0] / ovs_factor
+
+    return x_peak, y_peak
+
+
+def get_cslc(cslc_file, pol='VV'):
+    '''
+    Get CSLC-S1 array associated to 'pol'
+
+    Parameters
+    ----------
+    cslc_file: str
+        File path to CSLC-S1 product
+    pol: str
+        Polarization string of CSLC-S1 array
+        to extract from product
+
+    Returns
+    -------
+    cslc: np.ndarray
+        CSLC-S1 image associated to polarization
+        channel 'pol'
+    '''
+    with h5py.File(cslc_file, 'r') as h5:
+        cslc = h5[f'science/SENTINEL1/CSLC/grids/{pol}'][()]
+    return cslc
+
+
+def get_xy_info(cslc_file):
+    '''
+    Get X/Y-coordinate vector and spacings
+    from CSLC-S1 product in 'cslc_file'
+
+    Parameters
+    ----------
+    cslc_file: str
+        File path to CSLC-S1 product
+
+    Returns
+    -------
+    x_vect: np.ndarray
+        Array of X-coordinates associated to CSLC-S1
+    y_vect: np.ndarray
+        Array of Y-coordinates associated to CSLC-S1
+    x_spac: np.float
+        CSLC-S1 spacing along X-direction
+    y_spac: np.float
+        CSLC-S1 spacing along Y-direction
+    '''
+
+    with h5py.File(cslc_file, 'r') as h5:
+        x_vect = h5['science/SENTINEL1/CSLC/grids/x_coordinates'][()]
+        y_vect = h5['science/SENTINEL1/CSLC/grids/y_coordinates'][()]
```

Comment:
Please note that there is half pixels of difference in the upper left corner's coordinates when the raster is loaded as `HDF5` and `NETCDF`. Based on what I've seen, the coordinates as NETCDF raster made more sense becuase the number was a multiple of 30.0